Add skins support via SkinsRestorer plugin

owlnull opened the pull request:
Non-legacy skins support via SkinsRestorer plugin in spigot version

Commit:
Add skins support via SkinsRestorer plugin

Commit:
Fix NullPointerException in SkinsRestorerSkinUrlProvider.getSkinUrl during joins player without skin

Commit:
Made dependency SkinsRestorer compileOnly

mikeprimm commented:
This looks promising - let me find some time to review it, and I'll be happy to integrate it!

Commit:
Fix plugin crash caused by SkinsRestorer integration
SkinsRestorer integration now just shows warning instead of plugin crash when SkinsRestorer not installed and integration enabled

mikeprimm commented:
Looks good - thanks!
